### PR TITLE
Add more functions to support test

### PR DESCRIPTION
**Pull request by hanahmily**
These changes introduce some  functions to support query module testing

### CODE REVIEW

File: banyand/stream/stream_query.go [hunk 1]
```diff
@@ -39,8 +41,12 @@ type Query interface {
 }
 
 type Stream interface {
+	io.Closer
+	Write(value *streamv2.ElementValue) error
```
Comment (hanahmily) [inline banyand/stream/stream_query.go:45]: Once the write returns an empty error, indices and raw data are all there. @lujiajing1126

**Comment by hanahmily** [comment]
@liuhaoyang Please take a look, do these changes support your need?

File: banyand/stream/stream_query.go (hunk 1 above)
Comment (hanahmily) [reply banyand/stream/stream_query.go:45]: From your PR's CI failures, the stream tests are flaky. This new write function can fix this defect.

**Comment by lujiajing1126** [comment]
https://github.com/apache/skywalking-banyandb/blob/main/banyand/stream/service.go#L103

Seems issue with this line. I guess we have to setup `writeListener` after this loop?

**Comment by hanahmily** [comment]
> https://github.com/apache/skywalking-banyandb/blob/main/banyand/stream/service.go#L103
> 
> Seems issue with this line. I guess we have to setup `writeListener` after this loop?

Prune it.

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**